# Exploração inicial

Notebook destinado à análise exploratória inicial dos dados.

##  Heart Disease Dataset

**Integrantes:** Diana Vitória Silva Barbosa, Gabriel Sasuke Yamauchi de Souza, Paulo Sérgio Martins de Miranda Júnior e Ricardo Henrique Barbosa
**Orientador:** Diego Roberto Gonçalves de Pontes
**Disciplina:** Data Science e Big Data
**Nome:** Heart Disease Dataset
**Link:** https://www.kaggle.com/datasets/johnsmith88/heart-disease-dataset
**Contexto:** previsão da presença de doença cardíaca em pacientes com base em indicadores clínicos (idade, pressão arterial, colesterol, etc.)
**Variáveis Prditoras:** idade, sexo, cp (tipo de dor no peito), trestbps (pressão arterial em repouso), chol (colesterol), fbs (glicemia em jejum), restecg (eletrocardiograma em repouso), thalach (frequência cardíaca máxima), exang (angina induzida por exercício), oldpeak (depressão do segmento ST), slope (inclinação do segmento ST), ca (número de vasos principais), thal (talassemia)
**Variável-alvo:** target (0 = ausência, 1 = presença de doença)
**Tipo de problema:** classificação binária

In [2]:
!pip install kagglehub
import kagglehub
import pandas as pd

path = kagglehub.dataset_download("johnsmith88/heart-disease-dataset")
df = pd.read_csv(f"{path}/heart.csv")
df.head()

Using Colab cache for faster access to the 'heart-disease-dataset' dataset.


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,52,1,0,125,212,0,1,168,0,1.0,2,2,3,0
1,53,1,0,140,203,1,0,155,1,3.1,0,0,3,0
2,70,1,0,145,174,0,1,125,1,2.6,0,0,3,0
3,61,1,0,148,203,0,1,161,0,0.0,2,1,3,0
4,62,0,0,138,294,1,1,106,0,1.9,1,3,2,0


## Análise Estatística do Dataset

Análise descritiva das variáveis e da distribuição da variável-alvo.

In [3]:
df.describe()

,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
count,1025.000000,1025.000000,1025.000000,1025.000000,1025.00000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000,1025.000000
mean,54.434146,0.695610,0.942439,131.611707,246.00000,0.149268,0.529756,149.114146,0.336585,1.071512,1.385366,0.754146,2.323902,0.513171
std,9.072290,0.460373,1.029641,17.516718,51.59251,0.356527,0.527878,23.005724,0.472772,1.175053,0.617755,1.030798,0.620660,0.500070
min,29.000000,0.000000,0.000000,94.000000,126.00000,0.000000,0.000000,71.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,48.000000,0.000000,0.000000,120.000000,211.00000,0.000000,0.000000,132.000000,0.000000,0.000000,1.000000,0.000000,2.000000,0.000000
50%,56.000000,1.000000,1.000000,130.000000,240.00000,0.000000,1.000000,152.000000,0.000000,0.800000,1.000000,0.000000,2.000000,1.000000
75%,61.000000,1.000000,2.000000,140.000000,275.00000,0.000000,1.000000,166.000000,1.000000,1.800000,2.000000,1.000000,3.000000,1.000000
max,77.000000,1.000000,3.000000,200.000000,564.00000,1.000000,2.000000,202.000000,1.000000,6.200000,2.000000,4.000000,3.000000,1.000000


In [5]:
df.groupby('target')[['chol', 'thalach']].mean()

,chol,thalach
target,,
0,251.292585,139.130261
1,240.979087,158.585551


### Principais observações

- O dataset possui 1025 registros, sem valores nulos (todas as colunas têm count = 1025).
- A idade média dos pacientes é de aproximadamente 54,4 anos (desvio padrão de 9,1 anos),
  variando entre 29 e 77 anos, com mediana de 56 anos.
- A distribuição do target é bem equilibrada: 51,3% dos pacientes têm presença de doença
  cardíaca (target = 1) e 48,7% não têm (target = 0).
- A proporção de homens no dataset é de 69,6%, contra 30,4% de mulheres — ou seja, a
  amostra tem bem mais homens do que mulheres.
- O colesterol (chol) tem média de 246 mg/dl, com bastante variação entre os pacientes
  (desvio padrão de ~51,6, indo de 126 a 564).
- A frequência cardíaca máxima (thalach) tem média de 149 bpm, variando de 71 a 202.

- Comparando por grupo, pacientes com doença cardíaca (target = 1) apresentam frequência
  cardíaca máxima maior (158,6 bpm) do que pacientes sem doença (139,1 bpm), uma diferença
  de quase 20 bpm que sugere que thalach tem relação forte com o diagnóstico — quanto
  maior a frequência cardíaca máxima atingida, maior a chance de o paciente estar no
  grupo com doença. Já o colesterol médio entre os dois grupos é próximo (251,3 mg/dl
  para quem não tem a doença vs 240,9 mg/dl para quem tem), uma diferença pequena que
  indica uma relação mais fraca entre colesterol e a presença de doença cardíaca —
  inclusive, contraintuitivamente, o colesterol médio é um pouco menor no grupo com
  doença.